packages/shared/lib/core/wallet/utils/convertToRawAmount.ts

```diff
@@ -15,14 +16,15 @@ export function convertToRawAmount(amount: string, tokenMetadata: ITokenMetadata
     }
 }
 
-function convertToRawAmountFromMetadata(amount: number, tokenMetadata: ITokenMetadata, unit: string): Big {
-    if (tokenMetadata.useMetricPrefix) {
+function convertToRawAmountFromMetadata(amount: number, tokenMetadata: TokenMetadata, unit: string): Big {
+    const isBaseToken = tokenMetadata?.standard === TokenStandard.BaseToken
+    if (isBaseToken && tokenMetadata.useMetricPrefix) {
         return Big(amount * IOTA_UNIT_MAP?.[unit?.substring(0, 1)] ?? 0)
     } else {
-        if (unit && unit === tokenMetadata.unit) {
+        if (unit && isBaseToken && unit === tokenMetadata.unit) {
             const decimals = Math.min(tokenMetadata.decimals, MAX_SUPPORTED_DECIMALS)
             return Big(amount).mul(Big(10).pow(decimals))
-        } else if (unit === tokenMetadata.subunit) {
+        } else if (isBaseToken && unit === tokenMetadata.subunit) {
```

Comment:
we dont have any return here for when `!isBaseToken`, we could just `return Big(amount) `